### ISSUE TITLE
Flow not working anymore (NC 22.2.6.2, App 1.3.6)

### ISSUE THREAD

**Issue by MadMakz** [cwilby/nextcloud-workflow-media-converter#207]
Not sure when it started but my convertflow isn't triggered anymore and silently fails.

It somewehere happened starting in the past two either NC or mediaconverter patches (i guess the latter).
Debug log doesn't tell much.

**However**, i reverted back to App version 1.3.2 (since i definently know it worked back in January) and it works!

PHP 7.4.28, mariadb 15.1, nginx 1.18.0

My flow:
![flow](https://user-images.githubusercontent.com/3930214/160297825-e69caa3a-2d93-4230-8489-4bb2c516acc7.jpg)
Full regex is: `/^.+\.(?:mkv|webm|wmv|mpe?g|avi|vob|dat|mov)$/i`

**Comment by MadMakz** [comment]
Update: Breaks on 1.3.3+ IF "Enable parallel media conversation" is DISABLED.

**Comment by Lutwidse** [comment]
Having the same issue with Nextcloud 23.0.3 (PHP 8.0.17) docker container.
Batch jobs are stuck in "Converting" if parallel media conversation is disabled.
 As mentioned in #198 there seems a compatible issue with PHP 8+ but there is no problem with the conversion.
 ```
[PHP] Error: Undefined array key "convertMediaInParallel" at /var/www/html/custom_apps/workflow_media_converter/lib/BackgroundJobs/ConvertMediaJob.php#81
```
![image](https://user-images.githubusercontent.com/74334594/162610375-e969ffa6-6ee1-4fcc-a414-8e31e0956b83.png)

**Comment by Lutwidse** [comment]
> Hey all.
> 
> Just dropping in to say, "The Real World™️" is keeping me busy, but I'm aware of the issue.
> 
> Hoping to have a fix out in the next couple of days, but open to somebody contributing a fix as well.

Hi cwilby. I hope you're doing well.
Unfortunately, I've no experience with PHP ever but I love this project!
I'd like to tip some coffee if there's any way to get it to you.

**Comment by eraser2021999** [comment]
just my 2cents here, this is the content of the arguments array passed into ConvertMediaJob.php: 
/admin/files/20220421_111246 0447.mov;m4v;keep;;keep;;preserve;

Nextcloud Docker v.23

**Comment by rsimenok** [comment]
Also not working...
Nextcloud 23.0.4
Debian 5.10.106-1 (2022-03-17) i686 GNU/Linux
PHP 7.4.28

**Comment by cagnulein** [comment]
+1 :)

**Comment by cwilby** [comment]
Hi all, this should be resolved as of Version 1.6.4.

Note:  There is a "Enable Parallel Media Conversion" option under administration settings, that uses a locking system I implemented when unchecked.

That locking system is blocking any jobs from running, so it's best to enable parallel media conversion and bypass that while I work on fixing it.

And as always, be patient with Nextcloud's cron-based job system..
